pkg/pool-weighted/contracts/WeightedMath.sol

```diff
@@ -317,7 +374,10 @@ contract WeightedMath {
         uint256 taxableAmount = amountOutWithoutFee.mulUp(taxablePercentage);
         uint256 nonTaxableAmount = amountOutWithoutFee.sub(taxableAmount);
 
-        return nonTaxableAmount.add(taxableAmount.mulDown(FixedPoint.ONE.sub(swapFeePercentage)));
+        uint256 taxableAmountMinusFees = taxableAmount.mulDown(FixedPoint.ONE.sub(swapFeePercentage));
+
+        swapFee = taxableAmount - taxableAmountMinusFees;
+        amountOut = nonTaxableAmount.add(taxableAmountMinusFees);
```

Comment:
Taking inspiration from lines +194 to +197, this should be a little more readable:

```suggestion
        swapFee = taxableAmount.mulUp(swapFeePercentage);
        amountOut = nonTaxableAmount.add(taxableAmount.sub(swapFee));
```

Reply:
You're right, thanks!